lib/mastermind.rb

```diff
@@ -0,0 +1,54 @@
+class Mastermind
+  attr_reader :passcode, :input, :guess_attempts, :output
+  def initialize(passcode:, input: $stdin, output: $stdout, guess_attempts: 4)
+    @passcode = passcode
+    # @correct_answers = 0
+    @input = input
+    @guess_attempts = 5
```

Comment:
This method has a `guess_attempts` parameter, and it looks like it could be used here.